### DISCUSSION TITLE
Change commit message by config

### DISCUSSION THREAD

**Discussion by phnx47** [BetaHuhn/repo-file-sync-action#176] [Ideas]
My commits must be followed "Conventional Commits specification" in one repository. I can't merge it with the wrong format.

So, I need to change the commit message for some repositories.

If you don't have time I can try to implement it myself and raise PR

**Comment by BetaHuhn** [comment]
You can use the `COMMIT_PREFIX` input to specify a different commit prefix than the default `🔄`. This should be enough to get it working with the Conventional Commits specification.

**Reply by phnx47** [reply]
@BetaHuhn This isn't enough, with `COMMIT_PREFIX` message is "chore: Synced local [folder] with remote [folder]", but I need "chore: synced local [folder] with remote [folder]"

**Reply by BetaHuhn** [reply]
Makes sense. 

How would you imagine such an option to work? The action by default creates a single commit for each file that was changed, so the option would have to be more of a commit message template. Currently the message is also slightly different depending on if the file was changed or newly created.

A `COMMIT_MESSAGE_TEMPLATE` input might work. We could make a few variables available which would be replaced when the action runs. In your case something like `chore: synced local $SOURCE with remote $TARGET` with `$SOURCE` and `$TARGET` being replaced at runtime.

The other much simple option would be to simply make the commit message all lowercase.

Would you be able to create a PR for this @phnx47?

**Reply by phnx47** [reply]
@BetaHuhn Yep, something similar: https://github.com/adrianjost/files-sync-action#commit_message.

It isn't an emergency, I will work with this next two weeks

**Reply by phnx47** [reply]
Totally forgot about this, but I believe I will find time the next month. I want try implement `COMMIT_MESSAGE_TEMPLATE`!

**Reply by phnx47** [reply]
@BetaHuhn I see 3 types of messages: 

- `${ COMMIT_PREFIX } Synced file(s) with ${ GITHUB_REPOSITORY }`
- `${ COMMIT_PREFIX } Synced local '${ file.dest }' with remote '${ file.source }`
- `${ COMMIT_PREFIX } Created local '${ file.dest }' from remote '${ file.source }'`

It can be difficult create 1 template for 3 type of messages...

Can we start with just lower case? `Synced` -> `synced`, `Created` -> `created` 


Here explanation about [Subject Case in Conventional Commits](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional#subject-case).

**Reply by phnx47** [reply]
@BetaHuhn If you agree with lowercase I'll raise the PR

**Reply by BetaHuhn** [reply]
Sorry for the late response. Sounds good!
